### ISSUE TITLE
Compile Issue with high_dim_filter.so On Nvidia TX1: modified_permutohedral.cc error: 'feature' was not declared in this scope

### ISSUE THREAD

**Issue by kaisark** [sadeepj/crfasrnn_keras#45]
Hi. I ran into a modified_permutohedral.cc compile issue with Nvidia TX1 (Ubuntu 16). I am running Tensorflow 1.5 and Keras 2.2.2. I am compiling the gpu_version (USE_GPU := 1) with CUDA 8.

---------------------------------------------
modified_permutohedral.cc:359:22: error: 'feature' was not declared in this scope
   const float * f = (feature + k * num_dimensions);
                      ^
modified_permutohedral.cc: In member function 'void ModifiedPermutohedral::sseCompute(tensorflow::Tensor&, const tensorflow::Tensor&, int, bool, bool) const':
modified_permutohedral.cc:622:51: error: 'seqCompute_cpu' was not declared in this scope
  seqCompute_cpu( out, in, value_size, reverse, add);

-------------------------------
(cv40py35) nvidia@tegra-ubuntu:~/cviz/crfasrnn_keras/src/cpp$ make
nvcc -std=c++11 -c -o cudacode.o high_dim_filter.cu -I/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include -I/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/external/nsync/public -D_GLIBCXX_USE_CXX11_ABI=1  -x cu -Xcompiler -fPIC --expt-relaxed-constexpr -D FILTER_GPU=1
nvcc warning : The 'compute_20', 'sm_20', and 'sm_21' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(56): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(80): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(100): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(192): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(216): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(236): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/arena_impl.h(52): warning: integer conversion resulted in a change of sign

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/arena_impl.h(147): warning: integer conversion resulted in a change of sign

nvcc -std=c++11 -c -o modified_permutohedral.o modified_permutohedral.cu -I/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include -I/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/external/nsync/public -D_GLIBCXX_USE_CXX11_ABI=1  -x cu -Xcompiler -fPIC --expt-relaxed-constexpr -D FILTER_GPU=1
nvcc warning : The 'compute_20', 'sm_20', and 'sm_21' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(56): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(80): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(100): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(192): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(216): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(236): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/arena_impl.h(52): warning: integer conversion resulted in a change of sign

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/arena_impl.h(147): warning: integer conversion resulted in a change of sign

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(56): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(80): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(100): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(192): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(216): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(236): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/arena_impl.h(52): warning: integer conversion resulted in a change of sign

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/arena_impl.h(147): warning: integer conversion resulted in a change of sign

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/generated_message_reflection.h(689): warning: variable "unused" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(56): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(80): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(100): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(192): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(216): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/stubs/atomicops_internals_arm64_gcc.h(236): warning: variable "temp" was set but never used

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/arena_impl.h(52): warning: integer conversion resulted in a change of sign

/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/google/protobuf/arena_impl.h(147): warning: integer conversion resulted in a change of sign

g++ -std=c++11 -D_GLIBCXX_USE_CXX11_ABI=0 -shared -fPIC -I/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include -O2  -I/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow/include/external/nsync/public -o high_dim_filter.so high_dim_filter.cc modified_permutohedral.cc cudacode.o modified_permutohedral.o -L/home/nvidia/.virtualenvs/cv40py35/lib/python3.5/site-packages/tensorflow -ltensorflow_framework -L/usr/local/cuda/lib64 -lcuda -lcudart -D FILTER_GPU=1
modified_permutohedral.cc: In member function 'void ModifiedPermutohedral::init_cpu(const float*, int, int)':
modified_permutohedral.cc:359:22: error: 'feature' was not declared in this scope
   const float * f = (feature + k * num_dimensions);
                      ^
modified_permutohedral.cc: In member function 'void ModifiedPermutohedral::sseCompute(tensorflow::Tensor&, const tensorflow::Tensor&, int, bool, bool) const':
modified_permutohedral.cc:622:51: error: 'seqCompute_cpu' was not declared in this scope
  seqCompute_cpu( out, in, value_size, reverse, add);
                                                   ^
Makefile:50: recipe for target 'high_dim_filter.so' failed
make: *** [high_dim_filter.so] Error 1

**Comment by kaisark** [comment]
I was able to check out Master and compile successfully on AWS P2 instance (Ubuntu 16, tensorflow-gpu==1.8.0, Keras==2.0.8). I think there might be some typo issues with the GPU Branch of the project. When can we expect a full GPU version of the project? Thanks.

![labels](https://user-images.githubusercontent.com/3580312/46053802-45d59f00-c112-11e8-950c-c35df91ebf9e.png)
